build/dev-runtime.js

```diff
@@ -27,18 +27,16 @@ function Lifecycle() {
   return {
     start: () => {
       state.started = true;
-      state.error = undefined;
-      emitter.emit('started');
+      emitter.emit('message');
     },
     stop: () => {
       state.started = false;
     },
     error: error => {
-      state.error = error;
```

Comment:
I'm not super familiar with the order these lifecycle methods are called, but wouldn't we still want `state.error = error` here so if `lifecycle.error` is called before `lifecycle.wait`, the promise returned from `wait` will reject instead of registering a `message` listener?

Reply:
Yeah I'm not 100% sure either. I can change it back.